### PR TITLE
replace postcss-cssnext with postcss-preset-env, and get docs folder working on local & prod.

### LINKED ISSUE #1856: Magrate from cssnext to postcss-preset-env
The library cssnext is now deprecated and is replaced by postcss-preset-env.

The ReactToolbox should probably migrate to this library in order to stay up-to-date.
This may requires multiple modifications to current css. For example, the color() function will have to be replaced by the mod-color() function.

### PR DESCRIPTION
**Pull request by that1matt**
In this PR:

######  library changes:
- Replaced postcss-cssnext with postcss-preset-env as postcss-cssnext is deprecated. Resolves #1856 
- Updated all `color()` to the new `color-mod()`.

###### `./docs` directory changes:
- To get the local docs to work, I had to make a bunch of changes.
- Updated postcss-cssnext with postcss-preset-env
- Updated webpack 1 to webpack 4
- Restore some `commons.css` that was deleted as the 2.x beta.
- Updated `highlight-loader` to latest version, and worked around the libraries hard-coded value of `lang-`, by setting the `classPrefix` in the `markdown-loader`.
-- This should allow @javivelasco to delete the forked version `https://github.com/javivelasco/highlight-loader` and use the latest npm version available.
- I also updated some deps while I was there.

Also sorry for the huge 3rd commit, it took all of that to get the local docs working.

I tested building the static docs site, and I didn't notice any issues. 
Loading on dev also seems to work without any issues I can see.
All tests pass.

I really enjoy this repo, If you would ever like some help, I am more than happy to help out, lmk.

### CODE REVIEW

**Comment by that1matt** [comment]
@rubenmoya I was able to fix the issue with the `menus`, I forgot the config option in the root webpack config. I also fixed an issue with the markdown not appearing correctly for tables in the docs site. 

As a note: if this does get merged, the wiki migration guide will need to be updated to reflect these changes.

File: README.md [hunk 1]
```diff
@@ -234,11 +241,12 @@ const config = {
     postcss: () => {
     return [
       /* eslint-disable global-require */
-      require('postcss-cssnext')({
+      require('postcss-preset-env')({
+        stage: 0,
         features: {
-          customProperties: {
-            variables: reactToolboxVariables,
```
Comment (rubenmoya) [inline README.md]: Hi! I was giving this a last review before merging it and, aren't we losing the variables customization here?

Comment (that1matt) [reply README.md]: Yep, I went and added it back to the readme to showcase that the option is still there.

Comment (rubenmoya) [reply README.md]: Perfect! I'll try to merge the PR tonight

**Comment by rubenmoya** [comment]
Thank you for the PR!

**Pull request merged**